packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -111,88 +111,58 @@ export function useSelectableCollection(options: SelectableCollectionOptions): S
       return;
     }
 
+    const navigateToKey = (key: Key | undefined, childFocus?: FocusStrategy) => {
+      if (key != null) {
+        manager.setFocusedKey(key, childFocus);
+
+        if (e.shiftKey && manager.selectionMode === 'multiple') {
+          manager.extendSelection(key);
+        } else if (selectOnFocus) {
+          manager.replaceSelection(key);
+        }
+      }
+    };
+
     switch (e.key) {
       case 'ArrowDown': {
         if (delegate.getKeyBelow) {
           e.preventDefault();
           let nextKey = manager.focusedKey != null
-            ? delegate.getKeyBelow(manager.focusedKey)
-            : delegate.getFirstKey();
-
-          if (nextKey != null) {
-            manager.setFocusedKey(nextKey);
-            if (manager.selectionMode === 'single' && selectOnFocus) {
-              manager.replaceSelection(nextKey);
-            }
-          } else if (shouldFocusWrap) {
-            let wrapKey = delegate.getFirstKey(manager.focusedKey);
-            manager.setFocusedKey(wrapKey);
-            if (manager.selectionMode === 'single' && selectOnFocus) {
-              manager.replaceSelection(wrapKey);
-            }
-          }
-
-          if (e.shiftKey && manager.selectionMode === 'multiple') {
-            manager.extendSelection(nextKey);
```

Comment:
Here nextKey can be null/undefined, which means when pressing shift+arrowDown at the end of a multi-select list empties selection, which empties selection. It doesn't seem like an expected behavior and is changed in `navigateToKey`